packages/dds/map/src/test/directoryOracle.ts

```diff
@@ -133,27 +240,39 @@ export class SharedDirectoryOracle {
 
 			// Only validate keys that the oracle is tracking
 			// (keys loaded from snapshots may not fire events)
-			if (this.model.has(pathKey)) {
+			if (model.has(pathKey)) {
 				// Verify oracle has the correct value
 				assert.deepStrictEqual(
-					this.model.get(pathKey),
+					model.get(pathKey),
 					value,
-					`Value mismatch for key "${pathKey}": oracle=${this.model.get(pathKey)}, actual=${value}`,
+					`[${modelName}] Value mismatch for key "${pathKey}": oracle=${model.get(pathKey)}, actual=${value}`,
 				);
 			}
 		}
 
 		// Recursively validate subdirectories
 		for (const [, subdir] of dir.subdirectories()) {
-			this.validateDirectory(subdir);
+			this.validateDirectory(subdir, modelName, model);
```

Comment:
Somewhere around here maybe want to validate that the real directory and the model agree on what subdirectories exist (e.g. currently wouldn't catch if the model had an extra stale directory)?